src/config/indexer-config.c

```diff
@@ -0,0 +1,142 @@
+/*
+ * Wazuh Indexer Configuration
+ * Copyright (C) 2015, Wazuh Inc.
+ * August 31, 2023.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#ifndef CLIENT
+#ifndef WIN32
+
+#include "shared.h"
+#include "external/cJSON/cJSON.h"
+
+/* Global variables */
+static const char * special_array_keys[] = {
+    "indexer.hosts",
+    "indexer.ssl.certificate_authorities",
+    NULL
+};
+
+cJSON * indexer_config = NULL;
+
+bool indexer_config_is_special_array_key(const char * keypath)
+{
+    const char ** psearch = special_array_keys;
+    while(*psearch)
+    {
+        if(strcmp(keypath, *psearch) == 0)
+        {
+            return true;
+        }
+        psearch++;
+    }
+    return false;
+}
+
+int indexer_config_special_array_subnode_read(XML_NODE node, cJSON *output_json)
+{
+    int i;
+
+    if (!node)
+        return 0;
+
+    // Iterate over elements
+    for (i = 0; node[i]; i++) {
+        cJSON_AddItemToArray(output_json, cJSON_CreateString(node[i]->content));
+    }
+    return 0;
+}
+
+int indexer_config_subnode_read(const OS_XML *xml, XML_NODE node, cJSON *output_json, char * current_keypath)
+{
+    int i;
+    xml_node **children;
+    cJSON * subnode;
+    cJSON * existing_item;
+    cJSON * array_item;
+    char * subnode_keypath;
+    size_t subnode_keypath_len;
+
+    if (!node)
+        return 0;
+
+    // Iterate over elements
+    for (i = 0; node[i]; i++) {
+        subnode_keypath_len = strlen(current_keypath) + strlen(node[i]->element) + 2;
+        os_calloc(1, subnode_keypath_len, subnode_keypath);
+        snprintf(subnode_keypath, subnode_keypath_len, "%s.%s", current_keypath, node[i]->element);
+
+        if(indexer_config_is_special_array_key(subnode_keypath))
+        {
+            subnode = cJSON_CreateArray();
+            if((children = OS_GetElementsbyNode(xml, node[i])))
```

Comment:
I think that the current expression keeps the code cleaner since it is only one line.
Regarding the second expression you suggest, makes that the result of the first argument is discarded and then evaluated with the same value in the second argument. So, direct assignation and evaluation have better performance.

> In the C and C++ programming languages, the comma operator (represented by the token ,) is a binary operator that evaluates its first operand and discards the result, and then evaluates the second operand and returns this value (and type).

This is only about coding style.

Reply:
Thank you for the explanation!